examples/Main.hs

```diff
@@ -46,6 +52,53 @@ example2 = plam $ \x -> pmatch x $ \case
   PLeft n -> n + 1
   PRight n -> n - 1
 
+data SampleRecord f = SampleRecord {
+  sampleBool :: f PBool,
+  sampleInt :: f PInteger,
+  sampleString :: f PString}
+
+type instance ScottEncoded SampleRecord a = PBool :--> PInteger :--> PString :--> a
+
+newtype PRecord r s = PRecord{_getRecord :: r (Term s)}
```

Comment:
One issue I have with this is that the name is slightly confusing. Records should be defined in another way not using this mechanism, as that is simpler.

Reply:
The purpose of the newtype wrappers is to juggle the kinds: The record types like `SampleRecord` have a kind `((k -> Type) -> Type) -> Type` because the HKD pattern works with `(k' -> Type) -> Type`, but the `Term s`  wants a kind `k -> Type`. I also have a vague feeling that there should be a way to simplify all the kinds, but I don't have a concrete suggestion.

Reply:
Yeah, I get that, however I think users would be confused if they try using `PRecord` when they just want a record and not mutual recursion.